packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,345 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest', {
+  includeResourceTypes: [ec2.CfnVpc.resourceType]
+  }));
+// the only resource with tags will be the VPC
+
+// or tag all but the vpc with
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest', {
+  excludeResourceTypes: [ec2.CfnVpc.resourceType]
+  }));
+
+```
+
+#### Removing Tags
+
+Removing tags is accomplished by applying a `RemoveTag`. The important default
+is that remove tags override `Tag` unless specified differently.
+
+```ts
+import cdk = require('@aws-cdk/cdk');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest'));
+theBestStack.apply(new cdk.RemoveTag('StackType', 'TheBest'));
+// now nothing will be tagged
+```
+
+#### Tag Priority
+
+The final control you have is to set a priority number. Tags with highest
+priority number will always win. If tags have the same priority number the tag
+closest in tree distance to the resource will win.
+
+The default priority of `Tag` is 0. The default priority of `RemoveTag` is 1.
+
+```ts
+import cdk = require('@aws-cdk/cdk');
+import ec2 = require('@aws-cdk/aws-ec2');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest', {prioirty: 1}));
+
+const vpc = new ec2.VpcNetwork(marketingStack, 'MarketingVpc', {
+  maxAZs: 3, 
+});
+
+// Tag default is 0, so this has no affect
+vpc.apply(new cdk.Tag('StackType', 'TheBestVpc'));
+// RemoveTag default is 1, so this removes the tag from the VPC and children of
+// VPC (e.g. Subnet, RouteTable, NAT Gateway, etc)
+vpc.apply(new cdk.RemoveTag('StackType'));
+```
+
+You can use priority to control tagging regardless of the hierarchy of the tree.
+
+## Aspects
+
+Aspects are a mechanism to extend the CDK without having to directly impact the
+class hierarchy directly. We have implemented Aspects using the [Visitor
+Pattern](https://en.wikipedia.org/wiki/Visitor_pattern).
+
+### Aspects in the CDK
+
+An aspect in the CDK is defined by the interface:
+
+```ts
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * All aspects can visit an IConstruct
+   */
+  visit(node: IConstruct): void;
+}
+```
+
+Aspects can be applied to any `Construct`. During the tree
+preparation phase the aspect will visit each construct in the tree one
+time. Aspects are invoked in the order they were added to the `Construct`,
+starting at the `App` (root of the tree) and progressing in order to the leaf
+nodes (most commonly the CloudFormation Resource). Aspect authors will implement 
+the `visit(IConstruct)` function can inspect the `Construct` for specific characteristics.
+Such as, is this construct a CloudFormation Resource?
+
+### Tag and Remove Tag Aspect
```

Comment:
This should go to the Tagging section under an `API` heading or something like that